### PR TITLE
Define API for Betweenness Centrality

### LINKED ISSUE #2642: MNMG BC - Define C++ API
Define C++ API for MNMG BC

### LINKED ISSUE #2643: MNMG BC - Define C API
Define C API for MNMG BC

### LINKED ISSUE #2644: MNMG BC - Define C++ unit tests
Define C++ unit tests for MNMG BC

### LINKED ISSUE #2646: MNMG BC - Define C unit tests
Define C unit tests for MNMG BC

### PR DESCRIPTION
**Pull request by ChuckHastings**
Defines the C API, C++ API, C Unit tests and C++ unit tests for both SG and MG betweenness centrality.

Closes #2642 
Closes #2643 
Closes #2644 
Closes #2646

### CODE REVIEW

File: cpp/include/cugraph/algorithms.hpp [hunk 1]
```diff
@@ -303,6 +303,92 @@ void edge_betweenness_centrality(const raft::handle_t& handle,
...
+ * The current implementation does not support a weighted graph.
+ *
+ * If @p num_vertices and @p vertices are not specified then this function will compute
+ * exact betweenness (compute betweenness using a traversal from all vertices).
+ *
+ * If @p num_vertices or @p vertices are specified then we will compute approximate
+ * betweenness either by random sampling @p num_vertices as the seeds of the traversal, or
+ * by using the provided @p vertices as the seeds of the traversal.
+ *
+ * If both @p and @num_vertices are specified we will throw an error.
```
Comment (seunghwak) [inline cpp/include/cugraph/algorithms.hpp:321]: Should we better use `std::optional<std::variant>>`? Then, the code self documents the expected behavior and this comment is unnecessary.

File: cpp/include/cugraph/algorithms.hpp [hunk 2]
```diff
@@ -303,6 +303,92 @@ void edge_betweenness_centrality(const raft::handle_t& handle,
...
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @tparam edge_t Type of edge identifiers. Needs to be an integral type.
+ * @tparam weight_t Type of edge weights. Needs to be a floating point type.
+ * @tparam multi_gpu Flag indicating whether template instantiation should target single-GPU (false)
+ *
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param graph_view Graph view object.
+ * @param num_vertices Optional, if specified, how many vertices to randomly select
+ * @param vertices Optional, if specified this provides the list of vertices to app
```
Comment (seunghwak) [inline cpp/include/cugraph/algorithms.hpp:334]: It seems like some texts after `to app` got deleted.

File: cpp/include/cugraph/algorithms.hpp [hunk 3]
```diff
@@ -303,6 +303,92 @@ void edge_betweenness_centrality(const raft::handle_t& handle,
...
+ * pass through this edge. The weight parameter is currenlty not supported
+ *
+ * If @p num_vertices and @p vertices are not specified then this function will compute
+ * exact betweenness (compute betweenness using a traversal from all vertices).
+ *
+ * If @p num_vertices or @p vertices are specified then we will compute approximate
+ * betweenness either by random sampling @p num_vertices as the seeds of the traversal, or
+ * by using the provided @p vertices as the seeds of the traversal.
+ *
+ * If both @p and @num_vertices are specified we will throw an error.
```
Comment (seunghwak) [inline cpp/include/cugraph/algorithms.hpp:364]: Should we better use `std::optional<std::variant>>`? Then, the code self documents the expected behavior and this comment is unnecessary.

File: cpp/include/cugraph/algorithms.hpp [hunk 4]
```diff
@@ -303,6 +303,92 @@ void edge_betweenness_centrality(const raft::handle_t& handle,
...
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @tparam edge_t Type of edge identifiers. Needs to be an integral type.
+ * @tparam weight_t Type of edge weights. Needs to be a floating point type.
+ * @tparam multi_gpu Flag indicating whether template instantiation should target single-GPU (false)
+ *
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param graph_view Graph view object.
+ * @param num_vertices Optional, if specified, how many vertices to randomly select
+ * @param vertices Optional, if specified this provides the list of vertices to app
```
Comment (seunghwak) [inline cpp/include/cugraph/algorithms.hpp:377]: It seems like some texts after `to app` got deleted.

File: cpp/include/cugraph/algorithms.hpp (hunk 2 above)
Comment (ChuckHastings) [reply cpp/include/cugraph/algorithms.hpp]: Fixed missing text, cleaned up documentation based on using std::variant

File: cpp/include/cugraph/algorithms.hpp (hunk 3 above)
Comment (ChuckHastings) [reply cpp/include/cugraph/algorithms.hpp]: Didn't think of that.  Updated in next push.

File: cpp/include/cugraph/algorithms.hpp (hunk 4 above)
Comment (ChuckHastings) [reply cpp/include/cugraph/algorithms.hpp]: Updated in next push

**Review by seunghwak — APPROVED** [review]

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**